File: projects/project05/quad.rb
# quad.rb computes quadratic roots
#
# Input: a, b and c for quadratic equation
# Precondition: a, b and c are positive
# Output: roots of given input
#
# Created by: Mark Davis
# Date: March 7, 2016
#################################################################


# quad() computes quadratic roots
# Receive: a, b, c (floats)             
# PRE: a is positive
# Return: roots of given inputs
######################################################

def quad(a, b, c)
  results = Array.new #make array for storing results
  if a != 0 #if a is not 0
    answer = (b * b) - 4 * a * c #compute first part of quadratic equation
	if answer >= 0
	  results[0] = (-b + Math.sqrt(answer))/(2*a) #compute root 1
	  results[1] = (-b - Math.sqrt(answer))/(2*a) #compute root 2
  	else
	  print "*** quad() : b^2 - 4ac is negative!"; #output error message if negative
	  results[0] = 0 			#set values to 0
	  results[1] = 0
   	end
  else
    print "*** quad() : a is negative!"; 	#output error message if negative
    results[0] = 0 				#set values to 0
    results[1] = 0
  end  
  return results 				#return the array
end


def main
  print "Enter a: "; #prompt for a, b and c
  a = gets.to_f
  print "Enter b: ";
  b = gets.to_f
  print "Enter c: ";
  c = gets.to_f

  answer = quad(a, b, c)
  puts "Root 1 is: " << answer[0].to_s #output values
  puts "Root 2 is: " << answer[1].to_s
end

main

File: projects/project05/quad.java
/* quad.java the roots of the quadratic equation
 *
 * Input: a b c (double)
 * PRE: input is positive
 * Output: roots of given inputs
 * Completed by: Mark Davis March 7, 2016
 * Model Code: lab05 CS 214
 ********************************************************/
import java.util.Scanner; 
import java.math.*;

class quad {

	/***************************************************************
          quad() computes roots
    	* Receive: a b c (doubles)
   	* PRE: numbers are positive
    	* Return: root 1 and root 2
   	*/
	public static boolean quad(double a, double b, double c, double[] results) { //declare function
	  if (a != 0) {
		double answer = Math.pow(b, 2.0) - 4 * a * c; //compute answer
		if (answer >= 0) { //if bigger than 0
			results[0] = (-b + Math.sqrt(answer))/(2*a); //compute first root
			results[1] = (-b - Math.sqrt(answer))/(2*a); //compute second root
			return true;
		} else {
			System.err.println("\n*** quad(): b^2 - 4ac is negative!"); //else output error message
			results[0] = results[1] = 0; //set roots to 0
			return false;
		}
	  }
	  else {
		System.err.println("\n*** quad(): a is zero!"); //else output error message
		results[0] = results[1] = 0; //set roots to 0
		return false;	
	  }
	}

	public static void main(String[] args) {
		Scanner keyboard = new Scanner(System.in); //scan keyboard for input
		System.out.println("\nEnter a b and c: "); //prompt for input
		double a, b, c; //declare variables for a, b and c
		a = keyboard.nextDouble(); //store keyboard inputs
		b = keyboard.nextDouble();
		c = keyboard.nextDouble();
		double [] results = new double[2]; //make array for storing results
		quad(a, b, c, results); //call quad to compute roots
		System.out.println("\n Roots are: " + results[0] + results[1]); //display roots		
		
	}
}
